src/components/notification-manager.ts

```diff
@@ -96,14 +151,22 @@ export default class NotificationManager {
         notifications.forEach((notification: NotificationElement) => {
             if (notification.rule.type === 'ZOOM_CONSTRAINT') this.setupZoomContraint(notification)
             if (notification.rule.type === 'AREA_CONSTRAINT') this.setupInclusionAreaConstraint(notification)
-            if (notification.type === 'info') useStore().getMap().addControl(new NotificationBoxControl(notification, 1))
+            if (notification.type === 'info') {
+                this.infosNotificationControl = new NotificationBoxControl(this.notificationControl.div, notification, 1)
+                useStore().getMap().addControl(this.infosNotificationControl);
+            }
         })
     }
 
     setupZoomContraint(rule: NotificationElement) {
+        this.zoomNotificationControl = new NotificationBoxControl(this.notificationControl.div, rule, 3);
+        this.zoomNotificationControl.div.classList.add('disabled');
+        useStore().getMap().addControl(this.zoomNotificationControl)
+
         if (this.hasValidZoom(rule)) {
-            useStore().getMap().addControl(new NotificationBoxControl(rule, 3))
-            this.validZoomConstraint = false;
+            this.validZoomConstraint = false; 
+            this.zoomNotificationControl.div.classList.remove('disabled');
+            this.zoomNotificationControl.div.classList.add('fade-in');
```

Comment:
you can probably refactor all those lines in a `show()` and `hide()`methods from NotificationBoxControl.
this way you don't have to make the `div` public.

You could also create animation mixins in case it would be used in another places than NotificationBoxControl

Reply:
Done for all controls, it's clear with this way